app/scripts/metamask-controller.js

```
@@ -4959,6 +4966,78 @@ export default class MetamaskController extends EventEmitter {
+      // this is only available for seedless onboarding flow
+      throw new Error(
+        'This method is only available for seedless onboarding flow',
+      );
+    }
+
+    const releaseLock = await this.syncSeedlessGlobalPasswordMutex.acquire();
+    try {
+      // recover the current device password
+      const { password: currentDevicePassword } =
+        await this.seedlessOnboardingController.recoverCurrentDevicePassword({
+          globalPassword,
+        });
+
+      // use current device password to unlock the keyringController vault
+      await this.submitPassword(currentDevicePassword);
+
+      try {
+        // update seedlessOnboardingController to use latest global password
+        await this.seedlessOnboardingController.syncLatestGlobalPassword({
+          oldPassword: currentDevicePassword,
+          globalPassword,
+        });
+
+        // update vault password to global password
+        await this.keyringController.changePassword(globalPassword);
+
+        // check password outdated again skip cache to reset the cache after successful syncing
+        await this.seedlessOnboardingController.checkIsPasswordOutdated({
+          skipCache: true,
+        });
+      } catch (err) {
+        // lock app again on error after submitPassword succeeded
+        await this.setLocked();
+        throw err;
+      }
+    } finally {
+      releaseLock();
+    }
+  }
+
+  /**
+   * Checks if the seedless password is outdated.
+   *
+   * @param {boolean} skipCache - whether to skip the cache
+   * @returns {Promise<boolean | undefined>} true if the password is outdated, false otherwise, undefined if the flow is not seedless
+   */
+  async checkIsSeedlessPasswordOutdated(skipCache = false) {
+    const { firstTimeFlowType } = this.onboardingController.state;
+
+    if (firstTimeFlowType !== FirstTimeFlowType.social) {
+      // this is only available for seedless onboarding flow
+      return undefined;
```

May be we can throw an error here or log a warning?

---

@lwin-kyaw this is being used at every app init, so i'm returning undefined to let it be called normally